pkg/transformer/openshift/openshift.go

```diff
@@ -59,10 +63,94 @@ func getImageTag(image string) string {
 	}
 }
 
+// hasGitBinary checks if the 'git' binary is available on the system
+func hasGitBinary() bool {
+	_, err := exec.LookPath("git")
+	return err == nil
+}
+
+// getGitRemote gets git remote URI for the current git repo
+func getGitRemote(composeFileDir string, remote string) (string, error) {
+	cmd := exec.Command("git", "ls-remote", "--get-url", remote)
+	cmd.Dir = composeFileDir
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	url := strings.TrimRight(string(out), "\n")
+
+	if !strings.HasSuffix(url, ".git") {
+		url += ".git"
+	}
+
+	return url, nil
+}
+
+// getGitCurrentBranch gets current git branch name for the current git repo
+func getGitCurrentBranch(composeFileDir string) (string, error) {
+	cmd := exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD")
+	cmd.Dir = composeFileDir
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimRight(string(out), "\n"), nil
+}
+
+// getGitRemoteForBranch gets git remote for a branch
+func getGitRemoteForBranch(composeFileDir string, branch string) (string, error) {
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("git branch -r | grep %s", branch))
+	cmd.Dir = composeFileDir
+
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.Split(strings.Trim(string(out), "\n "), "/")[0], nil
+}
+
+// getComposeFileDir returns compose file directory
+func getComposeFileDir(inputFile string) (string, error) {
+	if strings.Index(inputFile, "/") != 0 {
+		workDir, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		inputFile = filepath.Join(workDir, inputFile)
+	}
+	return filepath.Dir(inputFile), nil
+}
+
+// getAbsBuildContext returns build context relative to project root dir
+func getAbsBuildContext(context string, composeFileDir string) (string, error) {
+	cmd := exec.Command("git", "rev-parse", "--show-prefix")
+	cmd.Dir = composeFileDir
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	prefix := strings.Trim(string(out), "\n")
+	return filepath.Join(prefix, context), nil
+}
+
 // initImageStream initialize ImageStream object
 func (o *OpenShift) initImageStream(name string, service kobject.ServiceConfig) *imageapi.ImageStream {
 	tag := getImageTag(service.Image)
 
+	var tags map[string]imageapi.TagReference
+	if service.Build != "" {
+		tags = map[string]imageapi.TagReference{}
```

Comment:
this condition becomes redundant, because L140 has done this assignment this already.